src/audio/sink_source_utils.c

```diff
@@ -72,4 +72,69 @@ int source_to_sink_copy(struct sof_source *source,
 	sink_commit_buffer(sink, INT_MAX);
 	return 0;
 }
+
+int sink_fill_with_silence(struct sof_sink *sink, size_t size)
+{
+	uint8_t *dst_ptr;
+	uint8_t *dst_begin;
+	uint8_t *dst_end;
+	size_t dst_size;
+	int ret;
+
+	if (!size)
+		return 0;
+	if (size > sink_get_free_size(sink))
+		return -ENOSPC;
+
+	ret = sink_get_buffer(sink, size,
+			      (void **)&dst_ptr,
+			      (void **)&dst_begin,
+			      &dst_size);
+	if (ret)
+		return ret;
+
+	dst_end = dst_begin + dst_size;
+	while (size) {
+		uint32_t dst_to_buf_overlap = (uintptr_t)dst_end - (uintptr_t)dst_ptr;
+		uint32_t to_fill = MIN(dst_to_buf_overlap, size);
+
+		ret = memset_s(dst_ptr, dst_to_buf_overlap, 0, to_fill);
+		assert(!ret);
+
+		size -= to_fill;
+		dst_ptr += to_fill;
+		if (to_fill == dst_to_buf_overlap)
+			dst_ptr = dst_begin;
+	}
+
+	sink_commit_buffer(sink, INT_MAX);
+	return 0;
+}
+
+int source_drop_data(struct sof_source *source, size_t size)
+{
+	uint8_t const *src_ptr;
+	uint8_t const *src_begin;
+	uint8_t const *src_end;
+	size_t src_size;
+	int ret;
+
+	if (!size)
+		return 0;
+	if (size > source_get_data_available(source))
+		return -EFBIG;
+
+	ret = source_get_data(source, size,
+			      (void const **)&src_ptr,
+			      (void const **)&src_begin,
+			      &src_size);
+	if (ret)
+		return ret;
+
+	source_release_data(source, INT_MAX);
+	return 0;
+}
+
+EXPORT_SYMBOL(sink_fill_with_silence);
 EXPORT_SYMBOL(source_to_sink_copy);
+EXPORT_SYMBOL(source_drop_data);
```

Comment:
please place each `EXPORT_SYMBOL()` under respective function

Reply:
@marcinszkudlinski let's keep "unresolved" until actually resolved or at least explained why you the existing version is preferred

Reply:
Well, I see it has been moved, it is now at the end of source_drop_data function, isn't it?

Reply:
@marcinszkudlinski sorry for not explaining clearly. **Every** `EXPORT_SYMBOL()` should **immediately** follow the respective function. Like
```
void f1()
{
}
EXPORT_SYMBOL(f1);

void f2()
{
}
EXPORT_SYMBOL(f2);
```
etc. See https://github.com/thesofproject/sof/actions/runs/12891822335/job/35944645332?pr=9787 :
> WARNING: EXPORT_SYMBOL(foo); should immediately follow its function/variable
> #87: FILE: src/audio/sink_source_utils.c:134:
> +EXPORT_SYMBOL(sink_fill_with_silence);